### PR TITLE
Add GraphQL to Docker Packaged API Scan

### PR DESCRIPTION
**Pull request by ricekot**
Requires zaproxy/zap-extensions#2573 for synchronous parsing when the API is used.

Signed-off-by: ricekot <ricekot@gmail.com>

### CODE REVIEW

File: docker/zap-api-scan.py [hunk 1]
```diff
@@ -228,14 +238,14 @@ def main(argv):
...
-        logging.warning('Format must be either \'openapi\' or \'soap\'')
+    if format != 'openapi' and format != 'soap' and format != 'graphql':
+        logging.warning('Format must be either \'openapi\', \'soap\', or \'graphql\'')
         usage()
         sys.exit(3)
 
     if running_in_docker():
         base_dir = '/zap/wrk/'
-        if config_file or generate or report_html or report_xml or report_json or report_md or progress_file or context_file or target_file:
+        if config_file or generate or report_html or report_xml or report_json or report_md or progress_file or context_file or target_file or schema_file:
```
Comment (ricekot) [inline docker/zap-api-scan.py:248]: "target_file" and "schema_file" will always be empty at this point in the program. Should they be removed?

Comment (psiinon) [reply docker/zap-api-scan.py:248]: Just the fact that they have been specified means the user will need to mount a local dir, so this is good.

**Comment by psiinon** [comment]
This will work with the weekly as that includes graphql. To work with the stable release we'll either need to include graphql with that or explicitly install the add-on in this script just in case.
Not a blocker for merging, just a reminder for us all ;)

File: docker/zap-api-scan.py (hunk 1 above)
Comment (thc202) [reply docker/zap-api-scan.py:248]: They are not being specified though, they are initialised with the actual values later. Not sure if best Python practice but declaring them later would be more clear.

**Comment by ricekot** [comment]
Updated:
- Use schema only when format is graphql
- Only allow URLs in target when format is graphql
- Declare target_file and schema_file later

File: docker/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 All notable changes to the docker containers will be documented in this file.
 
+### 2020-10-16
```
Comment (thc202) [inline docker/CHANGELOG.md:4]: Could you update the date?

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**